x-pack/plugins/secops/public/components/page/network/ip_overview/field_renderers.tsx

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiIcon, EuiToolTip } from '@elastic/eui';
+import numeral from '@elastic/numeral';
+import { FormattedRelative } from '@kbn/i18n/react';
+import { getOr } from 'lodash/fp';
+import React from 'react';
+import styled from 'styled-components';
+
+import {
+  AutonomousSystem,
+  Domain,
+  HostEcsFields,
+  IpOverviewData,
+  IpOverviewType,
+  Overview,
+} from '../../../../graphql/types';
+import { DefaultDraggable } from '../../../draggables';
+import { getEmptyTagValue } from '../../../empty_value';
+import { ExternalLinkIcon } from '../../../external_link_icon';
+import { PreferenceFormattedDate } from '../../../formatted_date';
+import { HostDetailsLink, ReputationLink, VirusTotalLink, WhoIsLink } from '../../../links';
+
+import { IpOverviewId } from './index';
+import * as i18n from './translations';
+
+const MoreDomains = styled(EuiIcon)`
+  margin-left: 5px;
+`;
+
+export const locationRenderer = (
+  fieldNames: string[],
+  data: IpOverviewData
+): React.ReactElement => {
```

Comment:
Is there any rules that we can tell whether to create a template with `stateless component`  or `pure from recompose`? I am confused between this two...

Reply:
I think the [consensus](https://medium.com/groww-engineering/stateless-component-vs-pure-component-d2af88a1200b) is to use `stateless components` for smaller, easy to render components, but to switch over to `pure components` for perf gains in rendering. This was all before `hooks` came out though... 

Thanks for commenting and pointing this out :) These renderers started off small, but it's pretty clear now that I should refactor them into `pure components`. My original thought was to get something simple and re-usable that we could plug into an `EuiDescriptionList` to make the `Host Details` UI refactor a bit easier, but I think field renderers like these may prove to be a bit more useful across the entire application (similar to all the new row renderers Frank & Andrew have put together).

Reply:
ah, I see. I should have checked the doc for `recompose`. So `recompose's pure` is doing the same thing as React's pureComponent? That would be easier to tell the difference!

Reply:
We generally:
1) prefer `SFC`s unless there's a need for lifecycle methods
2) prefer `pure` when writing `SFC`s, and when lifecycles (assuming pre-hooks code) are required, prefer extending from `React.PureComponent`, to (in both cases), prevent unnecessary re-renders via the "free" (as in, you don't have to write it yourself) shallow checking of props in `shouldComponentUpdate` provided by the implementation of `pure` and `PureComponent`